naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala

```diff
@@ -0,0 +1,95 @@
+package org.coursera.naptime
+
+import com.linkedin.data.schema.DataSchema
+import com.linkedin.data.schema.RecordDataSchema
+import org.coursera.naptime.model.Keyed
+import org.coursera.naptime.resources.TopLevelCourierCollectionResource
+import org.coursera.naptime.router2.Router
+import org.junit.Test
+import org.scalatest.junit.AssertionsForJUnit
+
+/**
+ * This test suite uses Courier to exercise advanced use cases for Naptime.
+ */
+object NestedMacroCourierTests {
+
+  class CoursesResource extends TopLevelCourierCollectionResource[String, Course] {
+    override def resourceName: String = "courses"
+
+    override implicit lazy val Fields: Fields[Course] = BaseFields.withRelated("instructors" -> ResourceName("instructors", 1))
+
+    var relatedFun: Option[String => List[String]] = None
```

Comment:
I made this a `var` instead of a `val` so that tests could optionally configure different behavior. +1 about making the names consistent, however.

Naptime's style doesn't work so well with Mockito / other mocking libraries. So this is sort of a fake implementation.